sdk/textanalytics/Azure.AI.TextAnalytics/src/HealthcareOperation.cs

```diff
@@ -0,0 +1,231 @@
+        /// The interval between status requests to the server.
+        /// The interval can change based on information returned from the server.
+        /// For example, the server might communicate to the client that there is not reason to poll for status change sooner than some time.
+        /// </param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> used for the periodical service calls.</param>
+        /// <returns>The last HTTP response received from the server.</returns>
+        /// <remarks>
+        /// This method will periodically call UpdateStatusAsync till HasCompleted is true, then return the final result of the operation.
+        /// </remarks>
+        public override ValueTask<Response<RecognizeHealthcareEntitiesResultCollection>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken = default) =>
+            this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+
+        /// <summary>
+        /// Calls the server to get updated status of the long-running operation.
+        /// </summary>
+        /// <param name="async">When <c>true</c>, the method will be executed asynchronously; otherwise, it will execute synchronously.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> used for the service call.</param>
+        /// <returns>The HTTP response received from the server.</returns>
+        private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken)
+        {
+            if (!_hasCompleted)
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(HealthcareOperation)}.{nameof(UpdateStatus)}");
+                scope.Start();
+
+                try
+                {
+                    Response<HealthcareJobState> update = async
+                        ? await _serviceClient.HealthStatusAsync(new Guid(Id), _top, _skip, _showStats, cancellationToken).ConfigureAwait(false)
+                        : _serviceClient.HealthStatus(new Guid(Id), _top, _skip, _showStats, cancellationToken);
+
+                    _response = update.GetRawResponse();
+
+                    if (update.Value.Status == JobStatus.Succeeded)
+                    {
+                        // we need to first assign a vaue and then mark the operation as completed to avoid race conditions
+                        _value = Transforms.ConvertToRecognizeHealthcareEntitiesResultCollection(update.Value.Results);
+
+                        NextLink = update.Value.NextLink;
+                        _hasCompleted = true;
+                    }
+                    else if (update.Value.Status == JobStatus.Failed)
+                    {
+                        _hasCompleted = true;
+                        _requestFailedException = new RequestFailedException("Request Failed Exception:" + update.Value.Results.Errors.ToString());
```

Comment:
I believe what we want here is the `JobMetadata.Errors` instead of the independent errors in the documents.

Also, FYI, if the class doesn't have an implementation of the `ToString` method, then .NET will print the name of the type and not the content. so if you do `update.Value.Errors.ToString` won't work. You will need to write a small method to correctly populate the `RequestFailedException`